kingfisher_scrapy/extensions.py

```diff
@@ -148,14 +148,17 @@ def spider_closed(self, spider, reason):
         if reason not in ('finished', 'sample') or spider.pluck or spider.keep_collection_open:
             return
 
-        data = {
-            'collection_source': spider.name,
-            'collection_data_version': spider.get_start_time('%Y-%m-%d %H:%M:%S'),
-            'collection_sample': str(bool(spider.sample)),
-        }
-
+        data = self._build_data_to_send(spider)
         return self._request(spider, 'end_collection_store', data['collection_source'], data)
 
+    def spider_error(self, failure, response, spider):
+        data = self._build_data_to_send(spider, response.request.meta['file_name'], response.request.url, failure)
+        return self._request(spider, 'create_file_error', response.request.url, data)
+
+    def item_error(self, item, response, spider, failure):
+        data = self._build_data_to_send(spider, item['file_name'], item['url'], failure)
+        return self._request(spider, 'create_file_error', item['url'], data)
```

Comment:
Otherwise, the `infix` in `_request` will be the same for `spider_error` and `item_error`.

```suggestion
        return self._request(spider, 'create_file_error', item['file_name'], data)
```

Reply:
Done.